yang/frr-zebra.yang

```diff
@@ -1985,4 +1989,63 @@ module frr-zebra {
   }
 
   // End interface model augmentation
+
+  augment "/frr-route-map:lib"
+    + "/frr-route-map:route-map"
+    + "/frr-route-map:entry"
+    + "/frr-route-map:match-condition"
+    + "/frr-route-map:condition-value" {
+    case ipv4-prefix-length {
+      when "./condition = 'ipv4-prefix-length' or
+            ./condition = 'ipv4-next-hop-prefix-length'";
+      leaf ipv4-prefix-length {
+        type uint8 {
+          range "0..32";
+        }
+      }
+    }
+    case ipv6-prefix-length {
+      when "./condition = 'ipv6-prefix-length'";
+      leaf ipv6-prefix-length {
+        type uint8 {
+          range "0..128";
+        }
+      }
+    }
+    case source-protocol {
+      when "./condition = 'source-protocol'";
+      leaf source-protocol {
+        type frr-route-types:frr-route-types;
+      }
+    }
+    case source-instance {
+      when "./condition = 'source-instance'";
+      leaf source-instance {
+        type uint8 {
+          range "0..255";
+        }
+      }
+    }
+  }
+
+  augment "/frr-route-map:lib"
+          + "/frr-route-map:route-map"
+          + "/frr-route-map:entry"
+          + "/frr-route-map:set-action"
+          + "/frr-route-map:action-value" {
+    case source-v4 {
+      when "./action = 'source'";
+      leaf source-v4 {
+        description "IPv4 address";
+        type inet:ipv4-address;
+      }
+    }
+    case source-v6 {
+      when "./action = 'source'";
+      leaf source-v6 {
+        description "IPv6 address";
+        type inet:ipv6-address;
+      }
+    }
+  }
```

Comment:
Are below protocol interaction commands not required to be taken care?

ip protocol bgp route-map 
  ROUTE-MAP  Route map name

Reply:
I think I missed this one, I was more interested in getting the route-map library part working as an example for other daemons.

Reply:
so you would be update the same PR or you want to have separate PR for this?